zerver/tests/test_presence.py

```diff
@@ -59,14 +59,14 @@ def test_activity(self, unused_mock: mock.Mock) -> None:
             result = self.client_get("/realm_activity/zulip/")
             self.assertEqual(result.status_code, 200)
 
-        self.assert_length(queries, 8)
+        self.assert_length(queries, 10)
 
         flush_per_request_caches()
         with queries_captured() as queries:
             result = self.client_get("/user_activity/iago@zulip.com/")
             self.assertEqual(result.status_code, 200)
 
-        self.assert_length(queries, 4)
+        self.assert_length(queries, 5)
```

Comment:
I suspect you can not need any new changes if you change this:
`    records = records.select_related("user_profile", "client").only(*fields)`
to
`    records = records.select_related("user_profile__bot_owner", "client").only(*fields)`

Can you try that?

Reply:
> reduced the number of queries from 14 to 10. Earlier the total number of queries were 8 but these additional two queries are to fetch the bot owner.

I made the change and the number of queries reduced from 10 to 8, but adding role column in human tab causes the query increase by 1.